mcodingbot/utils/peps.py

```diff
@@ -39,10 +41,23 @@ async def fetch_pep_info(self, bot: Bot) -> None:
                     )
                     for pep_id, pep in (await resp.json()).items()
                 }
+                self._pep_map = {
+                    k: f"{v.title} ({v.number})" for k, v in self._peps.items()
+                }
 
     def get(self, pep_number: int) -> PEPInfo | None:
         return self._peps.get(pep_number)
 
+    def search(
+        self, query: str, limit: int = 20
+    ) -> Generator[PEPInfo, None, None]:
+        res = fuzzy_search(query, self._pep_map, limit=limit)
+        return (
+            pep_info
+            for pep in res
+            if (pep_info := self.get(pep[2])) is not None
+        )
```

Comment:
Why not just use `yield` in a normal loop? Would be more readable imo (mainly because of the walrus)

Reply:
+1

Reply:
+1 for tragic

Reply:
A yield seems like a good idea.

```py
def search(...) -> Generator[PEPInfo, None, None]:
    res = ...
    for pep in res:
        if (pep_info := self.get(pep[2])):
            yield pep
```